internal/driver/metadata.go

```diff
@@ -266,5 +279,37 @@ func getSupportedFrameRateFormats(d *usbdevice.Device) (interface{}, error) {
 		}
 		r.FrameRateFormats = append(r.FrameRateFormats, format)
 	}
-	return r, nil
+	resultMap := make(map[string]result)
+	resultMap[d.Name()] = r
+	return resultMap, nil
+}
+
+func GetFrameRate(d *usbdevice.Device) (interface{}, error) {
+	streamParam, err := d.GetStreamParam()
+	if err != nil {
+		return v4l2.Fract{}, err
+	}
+	timePerFrame := streamParam.Capture.TimePerFrame
+	var fps v4l2.Fract
+	fps.Denominator = timePerFrame.Numerator
+	fps.Numerator = timePerFrame.Denominator
+	result := make(map[string]v4l2.Fract)
+	result[d.Name()] = fps
+	return result, nil
+}
+
+func getCompleteFormatMapEntry(key uint32) (string, bool) {
+	unsupported := map[uint32]string{
+		PixelFmtDepth:  PixelFmtDepthDesc,
+		PixelFmtUYVY:   PixelFmtUYVYDesc,
+		PixelFmtGrey8:  PixelFmtGrey8Desc,
+		PixelFmtGrey12: PixelFmtGrey12Desc,
+	}
```

Comment:
this could be in a var block at the top of the file, instead of defined every time in the function:

```go
var (
	unsupportedPixelFmts = map[uint32]string{
		PixelFmtDepth:  PixelFmtDepthDesc,
		PixelFmtUYVY:   PixelFmtUYVYDesc,
		PixelFmtGrey8:  PixelFmtGrey8Desc,
		PixelFmtGrey12: PixelFmtGrey12Desc,
	}

)